### ISSUE TITLE
Loss INF when reproducing Stage 1 pre-training of BLIP2 under FP16

### ISSUE THREAD

**Issue by ustcwhy** [salesforce/LAVIS#267]
Thanks for your wonderful work. I try to reproduce the BLIP-2 pre-training with DeepSpeed. 

I notice that the original scripts do not use fp16 for Q-former. I set it as fp16 for faster training speed. However, the optimization seems very unstable, especially for LM loss, which goes to INF in the first few iterations.

I wonder whether you meet the same issue while pre-training BLIP-2? Q-former must be set as float32?

I use 6 GPU and the default setting https://github.com/salesforce/LAVIS/blob/main/lavis/projects/blip2/train/pretrain_stage1.yaml
The vision backbone is eva_vit_g.

**Comment by ChantalMP** [comment]
Hi @ustcwhy, may I ask which code you used for stage 1 pre-training? Did you implement it yourself, or is there any resource that one should be aware of?

Thanks a lot in advance!

**Comment by ldfandian** [comment]
> Thanks for your wonderful work. I try to reproduce the BLIP-2 pre-training with DeepSpeed.
> 
> I notice that the original scripts do not use fp16 for Q-former. I set it as fp16 for faster training speed. However, the optimization seems very unstable, especially for LM loss, which goes to INF in the first few iterations.
> 
> I wonder whether you meet the same issue while pre-training BLIP-2? Q-former must be set as float32?
> 
> I use 6 GPU and the default setting https://github.com/salesforce/LAVIS/blob/main/lavis/projects/blip2/train/pretrain_stage1.yaml The vision backbone is eva_vit_g.

did you try fp32 and is it stable?
